vmdb/app/controllers/container_controller.rb

```diff
@@ -0,0 +1,360 @@
+        session[:exp_parms] = {:id => tree_node_id}
+        redirect_to :action => "explorer"
+      end
+      format.any {render :nothing => true, :status => 404}
+    end
+  end
+
+  # Tree node selected in explorer
+  def tree_select
+    @explorer = true
+    @lastaction = "explorer"
+    self.x_node = params[:id]
+    @nodetype, id = params[:id].split("_").last.split("-")
+
+    if x_tree[:type] == :containers_filter && TreeBuilder.get_model_for_prefix(@nodetype) != "Container"
+      search_id = @nodetype == "root" ? 0 : from_cid(id)
+      adv_search_build(vm_model_from_active_tree(x_active_tree))
+      session[:edit] = @edit              # Set because next method will restore @edit from session
+      listnav_search_selected(search_id) unless params.has_key?(:search_text) # Clear or set the adv search filter
+      if @edit[:adv_search_applied] &&
+        MiqExpression.quick_search?(@edit[:adv_search_applied][:exp]) &&
+        %w(reload tree_select).include?(params[:action])
+        self.x_node = params[:id]
+        quick_search_show
+        return
+      end
+    end
+
+    replace_right_cell
+  end
+
+  # Accordion selected in explorer
+  def accordion_select
+    @layout     = "explorer"
+    @lastaction = "explorer"
+    self.x_active_accord = params[:id]
+    self.x_active_tree   = "#{params[:id]}_tree"
+    replace_right_cell
+  end
+
+  private
+
+  def get_record_display_name(record)
+    record.name
+  end
+
+  # Get all info for the node about to be displayed
+  def get_node_info(treenodeid)
+    @show_adv_search = true
+    @nodetype, id = valid_active_node(treenodeid).split("_").last.split("-")
+    #resetting action that was stored during edit to determine what is being edited
+    @sb[:action] = nil
+    if x_node == "root" || TreeBuilder.get_model_for_prefix(@nodetype) == "MiqSearch"
+      typ = "Container"
+      process_show_list
+      @right_cell_text = _("All %s") % ui_lookup(:models => typ)
+    else
+      show_record(from_cid(id))
+      @right_cell_text = _("%{model} \"%{name}\"") % {:name => @record.name,
+                                                      :model => ui_lookup(:model => TreeBuilder.get_model_for_prefix(@nodetype))}
+    end
+
+    if @edit && @edit.fetch_path(:adv_search_applied, :qs_exp) # If qs is active, save it in history
+      x_history_add_item(:id     => x_node,
+                         :qs_exp => @edit[:adv_search_applied][:qs_exp],
+                         :text   => @right_cell_text)
+    else
+      x_history_add_item(:id => treenodeid, :text => @right_cell_text)  # Add to history pulldown array
+    end
+
+    # After adding to history, add name filter suffix if showing a list
+    unless @search_text.blank?
+      @right_cell_text += _(" (Names with \"%s\")") % @search_text
+    end
+  end
+
+  #set partial name and cell header for edit screens
+  def set_right_cell_vars(action)
+    name = @record ? @record.name.to_s.gsub(/'/,"\\\\'") : "" # If record, get escaped name
```

Comment:
@chessbyte this method can be used to set header for the right cells as @erinboyd enhances these screens, hence i left it in there. If you want i can remove them, let me know.